pymeasure/instruments/racal/racal1992.py

```diff
@@ -0,0 +1,363 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import logging
+import time
+
+from pymeasure.instruments import Instrument
+
+log = logging.getLogger(__name__)   # https://docs.python.org/3/howto/logging.html#library-config
+log.addHandler(logging.NullHandler())
+
+
+class ReturnValueError(Exception):
+    pass
+
+
+class Racal1992(Instrument):
+    """ Represents the Racal-Dana 1992 Universal counter
+
+    .. code-block:: python
+
+        from pymeasure.instruments.racal import Racal1992
+        counter = Racal1992("GPIB0::10")
+
+    This class should also work for Racal-Dana 1991, it has the same
+    product manual, as long as you don't use functionality that requires
+    channel B.
+    """
+
+    int_types = ['SF', 'RS', 'UT', 'MS', 'TA']
+    float_types = ['CK', 'FA', 'PA', 'TI', 'PH', 'RA',
+                   'MX', 'MZ', 'LA', 'LB', 'FC', 'RC', 'DT', 'GS']
+
+    channel_params = {
+        'A' : {                                                         # noqa
+            'coupling'      : { 'AC'   : 'AAC',  'DC'     : 'ADC' },    # noqa
+            'attenuation'   : { 'X1'   : 'AAD',  'X10'    : 'AAE' },    # noqa
+            'trigger'       : { 'auto' : 'AAU',  'manual' : 'AMN' },    # noqa
+            'impedance'     : { '50'   : 'ALI',  '1M'     : 'AHI' },    # noqa
+            'slope'         : { 'pos'  : 'APS',  'neg'    : 'ANS' },    # noqa
+            'filtering'     : { True   : 'AFE',  False    : 'AFD' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+        'B' : {                                                         # noqa
+            'coupling'      : { 'AC'        : 'BAC',  'DC'     : 'BDC' },    # noqa
+            'attenuation'   : { 'X1'        : 'BAD',  'X10'    : 'BAE' },    # noqa
+            'trigger'       : { 'auto'      : 'BAU',  'manual' : 'BMN' },    # noqa
+            'impedance'     : { '50'        : 'BLI',  '1M'     : 'BHI' },    # noqa
+            'slope'         : { 'pos'       : 'BPS',  'neg'    : 'BNS' },    # noqa
+            'input_select'  : { 'separate'  : 'BCS',  'common' : 'BCC' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+    }
+
+    operating_modes = {
+        'self_check'      : 'CK',    # noqa
+        'frequency_a'     : 'FA',    # noqa
+        'period_a'        : 'PA',    # noqa
+        'phase_a_rel_b'   : 'PH',    # noqa
+        'ratio_a_to_b'    : 'RA',    # noqa
+        'ratio_c_to_b'    : 'RC',    # noqa
+        'interval_a_to_b' : 'TI',    # noqa
+        'total_a_by_b'    : 'TA',    # noqa
+        'frequency_c'     : 'FC',    # noqa
+    }
+
+    def decode(v, allowed_types=None):
```

Comment:
Please mark this method as static method, as it does not require instance or class:
```suggestion
    @staticmethod
    def decode(v, allowed_types=None):
```

Reply:
Done.